app/models/host/base.rb

```diff
@@ -57,6 +57,8 @@ def self.taxonomy_conditions
     scope :no_location,     -> { rewhere(:location_id => nil) }
     scope :no_organization, -> { rewhere(:organization_id => nil) }
 
+    delegate :ssh_authorized_keys, :to => :owner, :allow_nil => true
```

Comment:
This one shouldn't delegate when the owner is an user group. Or alternatively, you can implement the method to fetch all ssh keys for all users in the user group.

```ruby
:if => ->(host) { host.owner_type == 'User' }
```

OR


```ruby
def ssh_authorized_keys
return owner.ssh_authorized_keys if owner_type == 'User'
owner.users.map(&:ssh_authorized_keys).flatten.uniq
end
```

Probably the 2nd option is better